packages/atomic/src/components/insight/atomic-insight-result-attach-to-case-action/atomic-insight-result-attach-to-case-action.tsx

```diff
@@ -0,0 +1,101 @@
+import {AttachToCase, buildAttachToCase, Result} from '@coveo/headless/insight';
+import {Component, Event, State, h, Element, EventEmitter} from '@stencil/core';
+import AttachIcon from '../../../images/attach.svg';
+import DetachIcon from '../../../images/detach.svg';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {IconButton} from '../../common/iconButton';
+import {ResultContext} from '../../search/result-template-components/result-template-decorators';
+import {InsightBindings} from '../atomic-insight-interface/atomic-insight-interface';
+
+export interface InsightResultAttachToCaseEvent {
+  controller: AttachToCase;
+  result: Result;
+}
+/**
+ * @internal
+ * The `atomic-insight-result-attach-to-case-action` component can be nested inside a `atomic-insight-result-actions` to render an interactive button that will emit an `atomic/insight/attachToCase/attach` or `atomic/insight/attachToCase/detach` JavaScript event, based on its current state, when clicked.
+ *
+ * @part result-action-container - The result action container
+ * @part result-action-button - The result action button
+ * @part result-action-icon - The result action icon
+ */
+@Component({
+  tag: 'atomic-insight-result-attach-to-case-action',
+  styleUrl: 'atomic-insight-result-attach-to-case-action.pcss',
+})
+export class AtomicInsightResultAttachToCaseAction
+  implements InitializableComponent<InsightBindings>
+{
+  @InitializeBindings() public bindings!: InsightBindings;
+  @ResultContext() private result!: Result;
+
+  @Element() host!: HTMLElement;
+  @State() public error!: Error;
+
+  public attachToCase!: AttachToCase;
+
+  @BindStateToController('attachToCase')
+  @State()
+  public attachToCaseState!: {};
+
+  @Event({
+    eventName: 'atomic/insight/attachToCase/attach',
+    composed: true,
+    cancelable: true,
+    bubbles: true,
+  })
+  private attach!: EventEmitter<InsightResultAttachToCaseEvent>;
+
+  @Event({
+    eventName: 'atomic/insight/attachToCase/detach',
+    composed: true,
+    cancelable: true,
+    bubbles: true,
+  })
+  private detach!: EventEmitter<InsightResultAttachToCaseEvent>;
+
+  public initialize() {
+    const caseId: string =
+      this.bindings.engine.state.insightCaseContext?.caseId || '';
+    this.attachToCase = buildAttachToCase(this.bindings.engine, {
+      options: {
+        result: this.result,
+        caseId: caseId,
+      },
+    });
+  }
+
+  private onClick() {
+    if (this.attachToCase.isAttached()) {
+      this.detach.emit({controller: this.attachToCase, result: this.result});
+    } else {
+      this.attach.emit({controller: this.attachToCase, result: this.result});
```

Comment:
same remark here, instead of passing `this.attachToCase`, pass `() => {this.attachToCase.attach()}`